### ISSUE TITLE
unexpected-keyword-arg: false positive and confusing error message with Uninferable

### ISSUE THREAD

**Issue by dseomn** [pylint-dev/pylint#9983]
### Bug description

This code is triggering unexpected-keyword-arg, I think because pylint can't infer the value of `name`:

https://github.com/dseomn/rock-paper-sand/blob/b0cf620cee0cd118eb1b7ee34d15f9257d0fb3be/rock_paper_sand/media_filter_test.py#L391
```python
registry = media_filter.Registry(**{f"{name}_factory": factory})
```

### Command used

```shell
pylint rock_paper_sand
```

### Pylint output

```python
rock_paper_sand/media_filter_test.py:391:19: E1123: Unexpected keyword argument 'Uninferable_factory' in constructor call (unexpected-keyword-arg)
```

### Expected behavior

 Since pylint can't tell if the keyword arg is valid or not, I'd expect it to not complain.

Even if it does have a finding for this, the `'Uninferable_factory'` in the error message is confusing, since (as far as I can tell) it's a combination of a value supplied by pylint (`Uninferable`) and the f-string in the code, but the actual code would never have a keyword arg with the name `Uninferable_factory`.

### Pylint version

https://github.com/dseomn/rock-paper-sand/actions/runs/11061313391/job/30733661830 shows pylint 3.3.1, astroid 3.3.4, and python 3.12.6.

**Comment by dseomn** [comment]
Seems to be working for me now.

**Comment by dseomn** [comment]
Or not. It worked locally but failed [under github actions](https://github.com/dseomn/rock-paper-sand/actions/runs/28055081595/job/83055256610).

Locally, where it worked:

```
$ pylint --version
pylint 4.0.5
astroid 4.1.1
Python 3.13.14 (main, Jun 10 2026, 18:10:12) [GCC 15.2.0]
```

GitHub Actions, where it failed:

```
Successfully set up CPython (3.13.14)
```

```
Successfully installed astroid-4.0.4 dill-0.4.1 isort-8.0.1 mccabe-0.7.0 pylint-4.0.6 tomlkit-0.15.0
```

**Comment by Pierre-Sassoulas** [comment]
My guess would be that the astroid version is not as capable in the ci. (4.0.4 vs 4.1.1 locally). Not sure why. What is the dep install command and operating system in CI ?

**Comment by dseomn** [comment]
Looks like `pip install pylint` on Ubuntu 24.04.4 LTS. pip and the python environment should come from actions/setup-python@v4 with `python-version: 3.13` instead of Ubuntu though.
